pkg/controller/sync/controller.go

```diff
@@ -431,6 +432,24 @@ func (s *FederationSyncController) reconcile(qualifiedName util.QualifiedName) u
 		if isSystemNamespace(s.fedNamespace, namespace) {
 			return util.StatusAllOK
 		}
+
+		// TODO(font): Consider how best to deal with qualifiedName keys for
+		// cluster-scoped template resources whose placement and/or override
+		// resources are namespace-scoped e.g. Namespaces. For now, update the
+		// Namespace template and placement keys depending on if we are
+		// reconciling due to a Namespace or FederatedNamespacePlacement
+		// update. This insures we will successfully retrieve the Namespace
+		// template or placement objects from the cache by using the proper
+		// cluster-scoped or namespace-scoped key.
+		if qualifiedName.Namespace == "" {
+			qualifiedName.Namespace = qualifiedName.Name
+			placementKey = qualifiedName.String()
+			glog.V(4).Infof("Received Namespace update for %v. Using placement key %v", key, placementKey)
+		} else if qualifiedName.Namespace != "" {
+			qualifiedName.Namespace = ""
+			key = qualifiedName.String()
+			glog.V(4).Infof("Received FederatedNamespacePlacement update for %v. Using template key %v", placementKey, key)
```

Comment:
Can you please show more detail why did the reconcile namespace can handle `FederatedNamespacePlacement` object?

Reply:
I'm not sure I understand your question. Can you restate?

Reply:
I'm clear for the logic now, `FederatedNamespacePlacement` is namespace scoped now, and the `qualifiedName` will include `Namespace`.